### DISCUSSION TITLE
How to use dataset "posetrack18" for 2D human pose detection

### DISCUSSION THREAD

**Discussion by DuKeM-CSGO** [open-mmlab/mmpose#1878] [Q&A] [answered]
According to [demo](https://mmpose.readthedocs.io/en/latest/demo.html#d-human-pose-bottom-up-image-demo), I can use dataset "coco" for detection easily. However, it seems there's not a neck keypoint in the detection result. That's why I want to use another dataset, such as "posetrack18":
```
python demo/top_down_img_demo.py \
        configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/posetrack18/hrnet_w48_posetrack18_384x288.py \
        hrnet_w48_posetrack18_384x288-5fd6d3ff_20211130.pth \
        --img-root tests/data/posetrack18 \
        --json-file tests/data/posetrack18/annotations/test_posetrack18_val.json \
        --out-img-root vis_results
```
And I got the result, yeah sure there's the neck keypoint within it. But I found this uses *annotation* data in `test_posetrack18_val.json`, which looks like been detected already, written into json and just used. So I can't change the source images at all, even I replace the demo images, I think the detection result is not right.

**Comment by ly015** [comment, marked as answer]
Hi, thanks for using MMPose. The json file (e.g. `test_posetrack18_val.json`) is only used to provide information like image paths and ground-truth bboxes, which is needed for a top-down pose model. And the keypoint result is predicted by the model.

To run inference on your own images, you would need to create a json file following the coco or posetrack18 format. The bbox information is required and the keypoint field can be left blank.

**Reply by DuKeM-CSGO** [reply]
Well thanks for answer, now I write my json and it seems to work. But I removed keys `mask_file` `ignore_regions_x` and `ignore_regions_y` in [demo](https://github.com/open-mmlab/mmpose/blob/master/tests/data/posetrack18/annotations/test_posetrack18_val.json#L647), and I modified `bbox_head` and `bbox` in [annotations](https://github.com/open-mmlab/mmpose/blob/master/tests/data/posetrack18/annotations/test_posetrack18_val.json#L1346) to [0, 0, 1920.0, 1080.0]. Then the result looks different at all(not fault at all though). Requiring the mask file and bboxes data here, how could I get them? Guess they're features of mmdet or others.

(Edit: Got the score of each keypoint. Sorry for the extra reply.)

**Comment by jin-s13** [comment]
You may try using mmdet for human detection. 
https://mmpose.readthedocs.io/en/latest/demo.html#using-mmdet-for-human-bounding-box-detection

Note mask is not necessary, but bboxes are needed.
